bundles/org.openhab.core/src/main/java/org/openhab/core/util/StringUtils.java

```diff
@@ -0,0 +1,305 @@
+        } else {
+            return str;
+        }
+    }
+
+    /**
+     * Simple method to escape XML special characters in String.
+     * There are five XML Special characters which needs to be escaped:
+     * 
+     * <pre>
+     * & - &amp;
+     * < - &lt;
+     * > - &gt;
+     * " - &quot;
+     * ' - &apos;
+     * </pre>
+     *
+     * @param xml the input xml as String to escape, may not be null
+     * @return the escaped xml as String
+     */
+    public static String escapeXml(String xml) {
+        xml = xml.replaceAll("&", "&amp;");
+        xml = xml.replaceAll("<", "&lt;");
+        xml = xml.replaceAll(">", "&gt;");
+        xml = xml.replaceAll("\"", "&quot;");
+        xml = xml.replaceAll("'", "&apos;");
+        return xml;
+    }
+
+    /**
+     * Capitalizes a String changing the first character to title case.
+     * No other characters are changed.
+     *
+     * <pre>
+     * ""       => ""
+     * "cat"    => "Cat"
+     * "cAt"    => "CAt"
+     * "'cat'"  => "'cat'"
+     * </pre>
+     *
+     * @param val the String to capitalize, may not be null
+     * @return the capitalized String
+     */
+    public static String capitalize(String val) {
+        if (val.length() == 0) {
+            return val;
+        }
+        return val.substring(0, 1).toUpperCase() + val.substring(1);
+    }
+
+    /**
+     * Capitalizes words in the string. Only the first char of every word is capitalized, other are set to lowercasing.
+     * Words are recognized by an underscore delimiter
+     *
+     * <pre>
+      * "openHAB_is_cool"   => "Openhab Is Cool"
+      * "foobar_Example" => "Foobar Example"
+     * </pre>
+     * 
+     * @param input the String to capitalize, may not be null
+     * @return the capitalized String
+     */
+    public static String capitalizeFully(String input) {
+        final String delimiter = "_";
+        String capitalizedFully = "";
+        for (String str : input.split(delimiter)) {
+            String properlyCapitalized = "";
+            if (str.length() > 0) {
+                properlyCapitalized = str.substring(0, 1).toUpperCase();
+            }
+            if (str.length() > 1) {
+                properlyCapitalized += str.substring(1).toLowerCase();
+            }
+            capitalizedFully = capitalizedFully + properlyCapitalized;
+        }
+
+        return capitalizedFully;
+    }
+
+    /**
+     * Capitalizes words in the string. Only the first char of every word is capitalized, the rest is left as is.
+     * Words are recognized by any whitespace.
+     *
+     * <pre>
+      * "openHAB is cool"   => "OpenHAB Is Cool"
+      * "foobar Example" => "Foobar Example"
+     * </pre>
+     * 
+     * @param input the String to capitalize, may be null
+     * @return the capitalized String
+     */
+    public static String capitalizeWords(@Nullable String input) {
+        String output = "";
+        if (input != null) {
```

Comment:
you may move the null check to the beginning to avoid creation of temporary objects

Reply:
done